### ISSUE TITLE
NotEquals operator behaves strange

### ISSUE THREAD

**Issue by matthiasvegh** [r0mai/tungsten#6]
The expression 5!=3 does not evaluate to false, rather it evaluates to 3. After this, the expression 5! evaluates to 3, implying that != is actually parsed as two tokens rather than one. Even if this is the case, 5! shouldn't be an expression that can be pattern matched. Perhaps the pattern matcher should be checked if the functioncall is protected or not.

**Comment by r0mai** [comment]
This is intended. Not equals operator is <>

**Comment by r0mai** [comment]
I think :D

**Comment by r0mai** [comment]
Or that is the concatenation operator?

**Comment by r0mai** [comment]
No, you were right, this processed incorrectly both syntactically and semantically. Factorial is a protected therefore it shouldn't be allowed to be assigned to.
